### PR TITLE
Add statistic pointcut

### PR DESCRIPTION
**Pull request by Mikhail-Pashkevich**

### CODE REVIEW

File: battle-service/src/main/java/by/pashkevich/mikhail/controller/BattleController.java [hunk 1]
```diff
@@ -23,40 +23,39 @@
...
 
+    @Statistic(message = "user create new battle")
     @PostMapping
     public BattleDto create(@Valid @RequestBody CreateDto createDto, @AuthenticationPrincipal User user) {
         Battle battle = battleService.create(createDto.getValue(), user);
 
         return battleMapper.toBattleDto(battle);
     }
 
+    @Statistic(message = "user join to any battle")
```
Comment (Volad) [inline battle-service/src/main/java/by/pashkevich/mikhail/controller/BattleController.java:40]: i wouldn't store it in controller.
but once we have save something saved in database.

File: battle-service/src/main/java/by/pashkevich/mikhail/client/StatisticClient.java [hunk 2]
```diff
@@ -1,10 +1,12 @@
 package by.pashkevich.mikhail.client;
 
+import by.pashkevich.mikhail.dto.StatisticDto;
 import org.springframework.cloud.openfeign.FeignClient;
 import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
 
 @FeignClient(name = "statisticClient", url = "${app.vars.statistic.url}")
```
Comment (Volad) [inline battle-service/src/main/java/by/pashkevich/mikhail/client/StatisticClient.java:8]: as far as i remember we talked about message broker usage and not rest template

File: battle-service/src/main/java/by/pashkevich/mikhail/controller/BattleController.java (hunk 1 above)
Comment (Mikhail-Pashkevich) [reply battle-service/src/main/java/by/pashkevich/mikhail/controller/BattleController.java:40]: Do you mean I should save query args every time I call the DB?

File: battle-service/src/main/java/by/pashkevich/mikhail/client/StatisticClient.java (hunk 2 above)
Comment (Mikhail-Pashkevich) [reply battle-service/src/main/java/by/pashkevich/mikhail/client/StatisticClient.java:8]: You are totally right.
But first and second PRs I just create statistic structure and understand what I should save.
After that I will implement message broker instead FeignClient

File: battle-service/src/main/java/by/pashkevich/mikhail/controller/BattleController.java (hunk 1 above)
Comment (Volad) [reply battle-service/src/main/java/by/pashkevich/mikhail/controller/BattleController.java:40]: no, i would better to formalize list of events with explicit data stracture , to send per specific action

**Review by Volad — APPROVED** [review]

**Pull request merged**